hoot-core-test/src/test/cpp/hoot/core/index/metric-hybrid/RFqHybridTreeTest.cpp

```diff
@@ -550,22 +550,22 @@ class RFqHybridTreeTest : public CppUnit::TestFixture
     OptimizeFunction ofDummy(keys, values, bounds);
     ofDummy.f(v);
 
-    OptimizeFunction* of = new OptimizeFunction(keys, values, bounds);
-    of->f(v);
+    OptimizeFunction of(keys, values, bounds);
+    of.f(v);
```

Comment:
This `OptimizeFunction` object used to never get cleaned up so I put it on the stack.